src/xrc/xh_chckb.cpp

```diff
@@ -43,7 +43,11 @@ wxObject *wxCheckBoxXmlHandler::DoCreateResource()
                     wxDefaultValidator,
                     GetName());
 
-    control->SetValue(GetBool( wxT("checked")));
+    if (GetBool("checked"))
+        control->SetValue(true);
+    else if (GetBool("undetermined") && (GetStyle() & wxCHK_3STATE))
+        control->Set3StateValue(wxCHK_UNDETERMINED);
```

Comment:
I think we should report an error if `undetermined` is used for a checkbox without 3 state style, it would be IMO more helpful than silently ignoring it. There is `ReportError()` which could be used for this.

What do you think?